lib/src/components/NxModal/__tests__/NxModal.test.tsx

```diff
@@ -33,17 +38,50 @@ describe('NxModal', function() {
 
   it('merges any passed in className to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal className="test"/>
+      <NxModal className="test" onClose={dummyCloseHandler}/>
     );
     const nxModalDiv = nxModal.find('.nx-modal');
     expect(nxModalDiv.hasClass('test')).toEqual(true);
   });
 
   it('includes any passed in attributes to the nx-modal div', function() {
     const nxModal = shallow(
-      <NxModal id="modal-id" data-foo="bar"/>
+      <NxModal id="modal-id" data-foo="bar" onClose={dummyCloseHandler}/>
     );
     expect(nxModal.find('.nx-modal').prop('id')).toEqual('modal-id');
     expect(nxModal.find('.nx-modal').prop('data-foo')).toEqual('bar');
   });
+
+  describe('NxModal event listener support', () => {
+    let container: HTMLDivElement | null;
+
+    it('executes onClose method when pressing ESC key', function () {
+      const mockCallBack = jest.fn();
+      const nxModal = (
+        <div>
+          <NxModal id="modal-id" onClose={mockCallBack}/>
+        </div>
+      );
+
+      act(() => {
+        mount(nxModal, {attachTo: container});
+      });
+
+      document.dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape'}));
+      expect(mockCallBack).toHaveBeenNthCalledWith(1);
+    });
+
+    beforeEach(function () {
+      // Rendering container for the component in test.
+      container = document.createElement('div');
+      document.body.appendChild(container);
+    });
+
+    afterEach(function () {
```

Comment:
`beforeEach` and `afterEach` should preferably go above the test itself

Reply:
> `beforeEach` and `afterEach` should preferably go above the test itself

Moved utility functions to the top of the describe: 8eff1da